### PR TITLE
Add runtime layer to run Bun on AWS Lambda

### LINKED ISSUE #352: Support AWS Lambda layer running bun
Hey

Is there a lambda layer for running bun instead of node?

### PR DESCRIPTION
**Pull request by Electroid**
Closes #352

### CODE REVIEW

File: packages/bun-lambda/runtime.ts [hunk 1]
```diff
@@ -0,0 +1,886 @@
...
+  if (value === null) {
+    exit(`Runtime failed to find the '${name}' environment variable`);
+  }
+  return value;
+}
+
+async function fetch(url: string, options?: RequestInit): Promise<Response> {
+  const { href } = new URL(
+    url,
+    `http://${env("AWS_LAMBDA_RUNTIME_API")}/2018-06-01/`
```
Comment (Jarred-Sumner) [inline packages/bun-lambda/runtime.ts]: can we move this string up to ensure it doesn't get recreated on each call

File: packages/bun-lambda/scripts/build.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  const tags = [
+    `${name}-${release.tag_name.replace("bun-v", "").replaceAll(".", "_")}`,
+  ];
+  if (latest.id === release.id) {
+    // TODO: tags.push(name);
+  }
+  for (const tag of tags) {
+    const cmd = [
+      "bunx",
+      "serverless",
```
Comment (Jarred-Sumner) [inline packages/bun-lambda/scripts/build.ts]: @thdxr do you have ideas on how to do this without adding `serverless` as a dependency?

Comment (thdxr) [reply packages/bun-lambda/scripts/build.ts]: Reposting what I sent on discord for record

1. create a zip
2. can use the AWS CLI to publisher that as a layer

if you don't want the AWS CLI as a dependency you could  use the AWS API to first upload the zip to an S3 bucket and then there's a lambda API that lets you publish a layer based off that file

Comment (ThatOneBro) [reply packages/bun-lambda/scripts/build.ts]: I think I have some old util scripts for uploading lambdas via s3 buckets, I can dig those up if this is our preferred method to do uploads

Comment (tobilg) [reply packages/bun-lambda/scripts/build.ts]: If you like, you can have a look at https://github.com/tobilg/duckdb-nodejs-layer/blob/main/publish_layer.sh

**Comment by lukeshay** [comment]
I am very interested in this! I made a [basic runtime](https://github.com/lukeshay/buntime) that mimics the functionality of the node runtime but I like the consistency of using the fetch API.

Does this work with non-HTTP events? (e.g. SQS, AppSync, Cognito)

**Comment by Electroid** [comment]
> I am very interested in this! I made a [basic runtime](https://github.com/lukeshay/buntime) that mimics the functionality of the node runtime but I like the consistency of using the fetch API.
> 
> Does this work with non-HTTP events? (e.g. SQS, AppSync, Cognito)

Yes, for non-HTTP events the payload will be in the `Request` body.

```ts
export default {
  async fetch(request: Request) {
    const event = await request.json();
  }
}
```

**Pull request merged**